File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw."
  [x y]
  (let [x (Math/pow (- x 0) 2)
        y (Math/pow (- y 0) 2)
        position (Math/sqrt (+ x y))]
    (cond
      (> position 10) 0
      (> position 5) 1
      (> position 1) 5
      :else 10)))

File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    (x * x + y * y)
    |> score_by_position()
  end

  defp score_by_position(pos) do
    cond do
      pos > 100 -> 0
      pos > 25 -> 1
      pos > 1 -> 5
      true -> 10
    end
  end
end
